applications/OptimizationApplication/python_scripts/algorithms/standardized_NLOPT_objective.py

```diff
@@ -0,0 +1,162 @@
+from KratosMultiphysics.OptimizationApplication.utilities.component_data_view import ComponentDataView
+from KratosMultiphysics.OptimizationApplication.utilities.logger_utilities import DictLogger
+from KratosMultiphysics.OptimizationApplication.utilities.logger_utilities import TimeLogger
+from KratosMultiphysics.OptimizationApplication.utilities.helper_utilities import CallOnAll
+import numpy
+import time as timer
+import datetime
+
+class StandardizedNLOPTObjective(ResponseRoutine):
+    """Standardized objective response function
+
+    This class transformed a user-given optimization problem into the standard format.
+    Supported objective types:
+        "minimization",
+        "maximization"
+
+    """
+    def __init__(self, parameters: Kratos.Parameters, master_control: MasterControl, optimization_problem: OptimizationProblem, required_buffer_size: int = 2):
+        default_parameters = Kratos.Parameters("""{
+            "response_name": "",
+            "type"         : "",
+            "scaling"      : 1.0
+        }""")
+        parameters.ValidateAndAssignDefaults(default_parameters)
+
+        response = optimization_problem.GetResponse(parameters["response_name"].GetString())
+
+        super().__init__(master_control, response)
+
+        if required_buffer_size < 2:
+            raise RuntimeError(f"Standardized objective requires 2 as minimum buffer size. [ response name = {self.GetReponse().GetName()} ]")
+
+        component_data_view = ComponentDataView(response, optimization_problem)
+        component_data_view.SetDataBuffer(required_buffer_size)
+
+        self.__optimization_problem = optimization_problem
+        self.__buffered_data = component_data_view.GetBufferedData()
+        self.__unbuffered_data = component_data_view.GetUnBufferedData()
+
+        scaling = parameters["scaling"].GetDouble()
+        if scaling < 0.0:
+            raise RuntimeError(f"Scaling should be always positive [ given scale = {scaling}]")
+
+        self.__objective_type = parameters["type"].GetString()
+        if self.__objective_type == "minimization":
+            self.__scaling = scaling
+        elif self.__objective_type == "maximization":
+            self.__scaling = -scaling
+        else:
+            raise RuntimeError(f"Requesting unsupported type {self.__objective_type} for objective response function. Supported types are: \n\tminimization\n\tmaximization")
+
+    def GetInitialValue(self) -> float:
+        if self.__unbuffered_data.HasValue("initial_value"):
+            return self.__unbuffered_data["initial_value"] * self.__scaling
+        else:
+            raise RuntimeError(f"Response value for {self.GetReponse().GetName()} is not calculated yet.")
+
+    def Initialize(self):
+        super().Initialize()
+
+    def CalculateStandardizedValueAndGradients(self, control_field: numpy.ndarray, gradient_field: numpy.ndarray, save_value: bool = True) -> float:
+
+        # update the master control
```

Comment:
Can you please comment whtat is happening here and why do you need it?

Reply:
@Igarizza I added more comments to the function, have a look to the new PR #11480